kernel/work.c

```diff
@@ -599,6 +602,50 @@ bool k_work_cancel_sync(struct k_work *work,
 	return pending;
 }
 
+#if defined(CONFIG_WORKQUEUE_WORK_TIMEOUT)
+static void workto_handler(struct _timeout *to)
+{
+	struct k_work_q *queue = CONTAINER_OF(to, struct k_work_q, workto);
+	k_spinlock_key_t key;
+	const char *name;
+	struct k_work *work;
+	k_work_handler_t handler;
+
+	key = k_spin_lock(&lock);
+	name = k_thread_name_get(&queue->thread);
+	work = queue->work;
+	handler = work->handler;
+	k_spin_unlock(&lock, key);
+
+	if (name != NULL) {
+		LOG_ERR("queue %s blocked by work %p with handler %p", name, work, handler);
+	} else {
+		LOG_ERR("queue %p blocked by work %p with handler %p", queue, work, handler);
+	}
```

Comment:
Still not crazy about this, as it needlessly duplicates a nearly identical string in the string table.

Reply:
How would you solve it? the string is about 30 bytes, and stored in ROM, the added operations of conditionally copying the name or a pointer to a RAM buffer to then pass to the LOG_ERR could easily take up the same or more ROM (while also adding complexity, the worst kind, involving null terminated strings)

Reply:
(LOG_ERR is probably adding code as well given the use of VARGS)

Reply:
> How would you solve it?

Using the code suggestion here:
https://github.com/zephyrproject-rtos/zephyr/pull/88345#discussion_r2036963133

Reply:
@bjarki-andreasen - can to address this comment as well?

Reply:
Done